app/web/features/userQueries/useLiteUsers.test.tsx

```diff
@@ -0,0 +1,283 @@
+
+      const { result, waitFor } = renderHook(
+        () => useLiteUsers(idsWithUndefined),
+        { wrapper }
+      );
+
+      await waitFor(() => expect(result.current.isLoading).toBe(false));
+
+      expect(mockGetLiteUsers).toHaveBeenCalledWith([1, 3]);
+    });
+
+    it("Should return an error if the query fails", async () => {
+      mockConsoleError();
+      const error = new RpcError(500, "Some error", {});
+      mockGetLiteUsers.mockRejectedValue(error);
+
+      const { result, waitForNextUpdate } = renderHook(
+        () => useLiteUsers([1, 2, 3]),
+        { wrapper }
+      );
+
+      await waitForNextUpdate();
+
+      expect(result.current.isLoading).toBe(false);
+      expect(result.current.error).toBe(error);
+      expect(result.current.data).toBeUndefined();
+    });
+
+    it("Should not run the query when ids is an empty array", async () => {
+      const ids: number[] = [];
+
+      renderHook(() => useLiteUsers(ids), { wrapper });
+
+      // Ensure the query function was NOT called
+      expect(mockGetLiteUsers).not.toHaveBeenCalled();
+    });
+
+    it("Should not run the query function when ids is undefined", () => {
+      const ids: (number | undefined)[] = [undefined];
+
+      renderHook(() => useLiteUsers(ids), { wrapper });
+
+      expect(mockGetLiteUsers).not.toHaveBeenCalled();
+    });
+  });
+
+  describe("useLiteUsers - cached data", () => {
+    const mockGetLiteUsers = service.user.getLiteUsers as jest.Mock;
+
+    const ids = [1, 2, 3, 4];
+
+    const sharedClient = new QueryClient({
+      defaultOptions: { queries: { retry: false } },
+    });
+
+    const sharedClientWrapper = ({
+      children,
+    }: {
+      children: React.ReactNode;
+    }) => (
+      <QueryClientProvider client={sharedClient}>
+        {children}
+      </QueryClientProvider>
+    );
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+
+      sharedClient.clear();
+      // Set the cached data for the specific liteUserIds
+      sharedClient.setQueryData(liteUsersKey(ids), getLiteUsers(ids));
+    });
+
+    it("Uses cached data instead of refetching", async () => {
+      const { result } = renderHook(() => useLiteUsers(ids), {
+        wrapper: sharedClientWrapper,
+      });
+
+      expect(mockGetLiteUsers).not.toHaveBeenCalled();
+
+      expect(result.current.isFetching).toBe(false);
+      expect(result.current.isLoading).toBe(false);
+    });
+
+    it("Should return cached data and set isError when fetching users fails", async () => {
+      mockConsoleError();
+      mockGetLiteUsers.mockRejectedValue(
+        new RpcError(StatusCode.INTERNAL, "Error fetching user data", {})
+      );
+
+      const { result, waitForNextUpdate } = renderHook(
+        () => useLiteUsers([1, 2, 3]),
+        { wrapper: sharedClientWrapper }
+      );
+
+      await waitForNextUpdate();
+
+      expect(result.current).toEqual(
+        expect.objectContaining({
+          data: undefined,
```

Comment:
Should this NOT be `undefined` based on the description of the test? 🤔

Reply:
I totally redid these tests after reading through the `useQuery` docs more. The way I had it wasn't actually using the cached data if there was an error as I had to set `retry: false` in the `useQuery` hook for that. In the new version I am checking that data is not empty though :-).